File: web/admin/src/request/CategoryPrompt.ts
import httpRequest, { ContentType, RequestParams } from './httpClient';

export interface CategoryPromptItem {
  id: string;
  name: string;
  content: string;
  /** 检索属性维度，逗号分隔 */
  attributes?: string;
}

export const getApiV1CategoryPrompts = (
  query: { id: string },
  params: RequestParams = {},
) =>
  httpRequest<{ items: CategoryPromptItem[] }>({
    path: '/api/v1/knowledge_base/category_prompts',
    method: 'GET',
    query,
    type: ContentType.Json,
    format: 'json',
    ...params,
  });

export const putApiV1CategoryPrompts = (
  body: { kb_id: string; items: CategoryPromptItem[] },
  params: RequestParams = {},
) =>
  httpRequest<unknown>({
    path: '/api/v1/knowledge_base/category_prompts',
    method: 'PUT',
    body,
    type: ContentType.Json,
    format: 'json',
    ...params,
  });

File: backend/domain/category_prompt.go
package domain

const SettingKeyCategoryPrompts = "category_prompts"

// CategoryPromptItem 后台「提示词」按品类维护的单条记录
type CategoryPromptItem struct {
	ID         string `json:"id"`
	Name       string `json:"name"`
	Content    string `json:"content"`
	Attributes string `json:"attributes"` // 检索属性维度，逗号分隔，用于附图命中品类后的属性提取引导
}

// CategoryPromptsReq 保存品类提示词列表（整表替换）
type CategoryPromptsReq struct {
	KBID  string               `json:"kb_id" validate:"required"`
	Items []CategoryPromptItem `json:"items"`
}
